### PR TITLE
[ADR] OIDC for JMAP IMAP and SMTP

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: src/adr/0050-oidc.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+## Status
+
+Accepted (lazy consensus).
+
+Not yet implemented.
+
+## Context
+
+Authentication is a critical item regarding security. THe overall internet community have been refining authentication
```
Comment (Arsnael) [inline src/adr/0050-oidc.md:13]:
```suggestion
Authentication is a critical item regarding security. The overall internet community have been refining authentication
```

File: src/adr/0050-oidc.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+### Handle OIDC within an Authentication strategy using JWKS and a backchannel for token invalidation
+
+Querying the JWKS endpoint, we would get the public key and use it to validate the JWT tokens.
+
+Also, token invalidation would need to be taken into account. An HTTP endpoint then needs to be used as a callback by 
+the OIDC provider to revoke tokens in each application (webadmin endpoints?). Revoked token would need to be stored, 
+and we likely would need an additional datastore for this use case (Redis?).
+
+Relying on the API gateway allows us not to worry about now about these two concerns.
```
Comment (Arsnael) [inline src/adr/0050-oidc.md:72]:
```suggestion
Relying on the API gateway allows us not to worry now about these two concerns.
```

File: src/adr/0050-oidc.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+### Handle OIDC within an Authentication strategy and call the OIDC userinfo endpoint
+
+The idea here is to fully delegate the token validation to the OIDC provider. The application server requests the OIDC 
+provider userinfo endpoint with the credentials given by the clients.
+
+Upsides: ease of implementation. Just an HTTP call, no crypto to be taking care of, no revokation to screw up.
+
+Downside: performance. Each request ends up calling the OIDC provider that then can be overloaded. This is definitely to 
+avoid with HTTP, but could make sense with IMAP/SMTP tht are longer lived connections.
```
Comment (Arsnael) [inline src/adr/0050-oidc.md:84]:
```suggestion
avoid with HTTP, but could make sense with IMAP/SMTP that are longer lived connections. 
```

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
(squashed)

**Review by Arsnael — APPROVED** [review]

**Pull request merged**